src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseSearchValuesTests.cs

```diff
@@ -0,0 +1,1023 @@
+                    private ReadOnlySpan<char> LongReadOnlySpanOfCharFromConstToCharArrayProperty => LongConstStringTypeMember.ToCharArray();
+                    private ReadOnlySpan<char> LongReadOnlySpanOfCharRVAPropertyWithGetAccessor1
+                    {
+                        get => new[] { 'a', 'e', 'i', 'o', 'u', 'A' };
+                    }
+                    private ReadOnlySpan<char> LongReadOnlySpanOfCharRVAPropertyWithGetAccessor2
+                    {
+                        get
+                        {
+                            return new[] { 'a', 'e', 'i', 'o', 'u', 'A' };
+                        }
+                    }
+                    private ReadOnlySpan<char> LongReadOnlySpanOfCharRVAPropertyWithGetAccessor3
+                    {
+                        get
+                        {
+                            Console.WriteLine("foo");
+                            return new[] { 'a', 'e', 'i', 'o', 'u', 'A' };
+                        }
+                    }
+
+                    public Test()
+                    {
+                        LongReadonlyCharArrayFieldWithSimpleFieldModification = new[] { 'a' };
+                        Console.WriteLine(InstanceReadonlyCharArrayField);
+                    }
+
+                    public void Dummy()
+                    {
+                        LongReadonlyCharArrayFieldWithSimpleElementModification[0] = 'a';
+                        Console.WriteLine(InstanceReadonlyCharArrayField[0]);
+                    }
+
+                    private void TestMethod(string str, ReadOnlySpan<char> chars, ReadOnlySpan<byte> bytes)
+                    {
+                        const string ShortConstStringLocal = "foo";
+                        const string LongConstStringLocal = "aeiouA";
+                        string NonConstStringLocal = "aeiouA";
+
+                        _ = chars.IndexOfAny("aeiou");
+                        _ = chars.IndexOfAny([|"aeiouA"|]);
+                        _ = chars.IndexOfAny("aeiouA" + NonConstStringProperty);
+                        _ = chars.IndexOfAny("aeiouA" + NonConstStringLocal);
+
+                        _ = chars.IndexOfAny(new[] { 'a', 'e', 'i', 'o', 'u' });
+                        _ = chars.IndexOfAny([|new[] { 'a', 'e', 'i', 'o', 'u', 'A' }|]);
+                        _ = chars.IndexOfAny([|new char[] { 'a', 'e', 'i', 'o', 'u', 'A' }|]);
+                        _ = chars.IndexOfAny(new[] { 'a', 'e', 'i', 'o', 'u', NonConstChar });
+
+                        _ = chars.IndexOfAny([|new[] { 'a', 'e', /* Comment */ 'i', 'o', 'u', 'A' }|]);
+                        _ = chars.IndexOfAny([|new[]
+                        {
+                            // Comment
+                            'a', 'e', 'i', 'o', 'u', 'A'
+                        }|]);
+
+                        _ = str.IndexOfAny([|new char[] { }|]);
```

Comment:
When the method in question is the `string.IndexOfAny(char[])` overload, an argument like `new char[] { ... }` will result in an allocation for every call, thus it's always worth replacing perf-wise.

For the `Span.IndexOfAny(ReadOnlySpan<char>)` overload, Roslyn will avoid the allocation, so it may very well already be okay perf-wise.

I've tried to only flag cases where we can actually expect the replaced code to perform better. Is that sort of behavior acceptable for an analyzer? (I agree it's likely worth pointing out on the analyzer doc at least)

Reply:
> I've tried to only flag cases where we can actually expect the replaced code to perform better. Is that sort of behavior acceptable for an analyzer? (I agree it's likely worth pointing out on the analyzer doc at least)

Sure it is, and yes, documenting would be helpful